app/main.py

```diff
@@ -1 +1,32 @@
-# write your code here
+from typing import List
+
+
+class Animal:
+    alive: List["Animal"] = []
+
+    def __init__(self, name: str, health: int = 100,
+                 hidden: bool = False) -> None:
+        self.name: str = name
+        self.health: int = health
+        self.hidden: bool = hidden
+        self.__class__.alive.append(self)
+
+    def __str__(self) -> str:
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, Hidden: {self.hidden}}}")
+
+    def __repr__(self) -> str:
+        return str(self)
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    def bite(self, target: Animal) -> None:
+        if isinstance(target, Herbivore) and not target.hidden:
+            target.health -= 50
+            if target.health <= 0:
+                self.__class__.alive.remove(target)
```

Comment:
Please, don't use `__class__` in public function, because it is private attribute. You can rewrite it without class
```suggestion
                self..alive.remove(target)
```